TF_Service_dotNet/TouchFree/Connections/MessageValidation.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Ultraleap.TouchFree.Library.Configuration;
+
+namespace Ultraleap.TouchFree.Library.Connections
+{
+    public static class MessageValidation
+    {
+        private static readonly Dictionary<string, Type> configJsonPropertyTypes = new()
+        {
+            { "interaction", typeof(InteractionConfig) },
+            { "physical", typeof(PhysicalConfig) }
+        };
+
+        /// <summary>
+        /// Checks for invalid states of the config request
+        /// </summary>
+        /// <param name="contentObj">JObject of content</param>
+        /// <returns>Returns a response as to the validity of the _content</returns>
+        public static Result<Empty> ValidateConfigJson(JObject contentObj)
+        {
+            bool anyValidContentFound = false;
+            foreach (var contentElement in contentObj)
```

Comment:
This doesn't currently handle any unexpected elements in the config json. At the moment we would return an error response if there is additional content at any level that we are not expecting.

e.g. "touch" in the base level

Reply:
This is deliberate. From a backwards compatibility standpoint being less strict is beneficial as we can remove parts of the message structure without causing it to fail validation. Clients can continue to send data that is removed in newer versions of the protocol to maintain backwards compatibility if required.

The validation scheme I imagine is most flexible and would like to standardize on amongst message queues is:
- Check always required fields are present
- Check for optional fields
- Check contextual constraints e.g. "at least one in set of n fields is required", "if x exists y is also required", "if k exists j must not exist"
- Ignore unexpected fields

Alternatively we could version message queue handlers and be strict about unexpected fields. This is more flexible long term but a lot more work that I wouldn't want to take on right now.

Reply:
After discussion last week - I've added functionality to fail validation when there's excess content fields in a config message and added a test case.